frontend/src/components/object/ObjectList.vue

```diff
@@ -43,7 +45,10 @@ const closeUpload = () => {
 
 const listObjects = async () => {
   try {
-    await objectStore.listObjects({ bucketId: route.query.bucketId });
+    await Promise.all([
+      bucketStore.getBucketPermissionsForUser(route.query.bucketId?.toString() || ''),
+      objectStore.listObjects({ bucketId: route.query.bucketId })
```

Comment:
So if you directly navigate to a object list landing page, you will have your perms for that bucket populated.